### PR TITLE
Added user-facing log about degraded mode.

### PR DESCRIPTION
**Pull request by sawsa307**
* Emit events when error state is set and reset to notify error state status of the syncer.
* When error state is set, degraded mode calculation is used only if "--enableDegradedMode" flag is turned on. No behavioral change is triggered by error state if the flag is not turned on.
* Error state is reset only if "--enableDegradedMode" flag is on and there is no difference between normal and degraded mode calculation.

Example logs:
```
I1130 00:06:03.754769 3742241 event.go:307] "Event occurred" object="default/neg-demo-svc" fieldPath="" kind="Service" apiVersion="v1" type="Normal" reason="SetErrorState" message="Set error state for NEG default/neg-demo-svc-k8s1-89f6839e-default-neg-demo-svc-80-b0c7e439-http/80-9376-GCE_VM_IP_PORT-L7 due to endpoint information for attach operation is incorrect"

I1130 00:07:17.248930 3742241 event.go:307] "Event occurred" object="default/neg-demo-svc" fieldPath="" kind="Service" apiVersion="v1" type="Normal" reason="ResetErrorState" message="Reset error state for NEG default/neg-demo-svc-k8s1-89f6839e-default-neg-demo-svc-80-b0c7e439-http/80-9376-GCE_VM_IP_PORT-L7"
```

### CODE REVIEW

File: pkg/neg/syncers/transaction.go [hunk 1]
```diff
@@ -231,7 +231,8 @@ func (s *transactionSyncer) syncInternal() error {
 	err := s.syncInternalImpl()
 	if err != nil {
 		if syncErr := negtypes.ClassifyError(err); syncErr.IsErrorState {
-			s.logger.Info("Setting error state", "error state", syncErr.Reason)
+			s.logger.Info("Set error state", "error state", syncErr.Reason)
+			s.recordEvent(apiv1.EventTypeWarning, "SetErrorState", fmt.Sprintf("Set error state for NEG %s due to %v", s.NegSyncerKey.String(), syncErr))
```
[the remark this answers is not in the document]
Comment (sawsa307) [reply pkg/neg/syncers/transaction.go:236]: Updated!

File: pkg/neg/syncers/transaction.go [hunk 2]
```diff
@@ -313,7 +314,8 @@ func (s *transactionSyncer) syncInternalImpl() error {
 			targetMap = degradedTargetMap
 			endpointPodMap = degradedPodMap
 			if len(notInDegraded) == 0 && len(onlyInDegraded) == 0 {
-				s.logger.Info("Resetting error state")
+				s.logger.Info("Reset error state")
+				s.recordEvent(apiv1.EventTypeNormal, "ResetErrorState", fmt.Sprintf("Reset error state for NEG %s", s.NegSyncerKey.String()))
```
[the remark this answers is not in the document]
Comment (sawsa307) [reply pkg/neg/syncers/transaction.go]: Updated!

File: pkg/neg/syncers/transaction.go [hunk 3]
```diff
@@ -552,7 +554,8 @@ func (s *transactionSyncer) operationInternal(operation transactionOp, zone stri
 		if syncErr.IsErrorState {
 			s.logger.Error(err, "Detected unexpected error when checking endpoint update response", "operation", operation)
 			s.syncLock.Lock()
-			s.logger.Info("Setting error state", "error state", syncErr.Reason)
+			s.logger.Info("Set error state", "error state", syncErr.Reason)
+			s.recordEvent(apiv1.EventTypeNormal, "SetErrorState", fmt.Sprintf("Set error state for NEG %s due to %v", s.NegSyncerKey.String(), syncErr))
```
[the remark this answers is not in the document]
Comment (sawsa307) [reply pkg/neg/syncers/transaction.go:563]: For this one, do you mean we need to change s.logger message  or set/resetErrorState function?

**Review by swetharepakula — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**